src/overlays/actors/ovl_En_Famos/z_en_famos.c

```diff
@@ -28,133 +62,810 @@ const ActorInit En_Famos_InitVars = {
+        rock->rotation.y += (s16)(((u32)Rand_Next() >> 0x17) + 0x900);
+        rock->rotation.z += (s16)(((u32)Rand_Next() >> 0x17) + 0xB00);
+    }
+}
+
+void EnFamos_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnFamos* this = THIS;
+    f32 oldHeight;
+    s32 oldHoverTimer; // save old value to test if changed
+
+    if (this->debrisTimer <= 0 ||
+        (this->debrisTimer--, EnFamos_UpdateDebrisPosRot(this), (this->actionFunc != EnFamos_DeathFade))) {
+        oldHoverTimer = this->hoverTimer;
+        EnFamos_UpdateFlipStatus(this);
+        if (this->cratorDespawnTimer > 0) {
+            this->cratorDespawnTimer--;
+            if (this->cratorDespawnTimer == 0) {
+                this->actor.child->parent = NULL; // child is EnTest (Crator after stomp)
+            }
+        }
+
+        this->actionFunc(this, globalCtx);
+        oldHeight = this->actor.world.pos.y;
+        Actor_MoveWithoutGravity(&this->actor);
+        if (oldHoverTimer != this->hoverTimer) { // test if updated in actionFunc
+            this->baseHeight += this->actor.world.pos.y - oldHeight;
+        }
+
+        if (this->flippedTimer >= 0) {
+            Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 35.0f, 30.0f, 80.0f, 0x1F);
+            if ((this->actionFunc == EnFamos_Attack) && (this->animatedMaterialIndex != FAMOS_ANIMATED_MAT_NORMAL) &&
+                (this->actor.bgCheckFlags & 1)) { // touch floor
+                this->actor.world.pos.y -= 60.0f;
+            }
+        }
+
+        this->actor.focus.rot.y = this->actor.shape.rot.y;
+        Collider_UpdateCylinder(&this->actor, &this->collider1);
+        if (this->collider1.base.atFlags & AT_ON) {
+            CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->collider1.base);
+        }
+
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider1.base);
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider1.base);
+        if (this->emblemCollider.base.acFlags & AC_ON) {
+            CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->emblemCollider.base);
+        }
+
+        if (this->collider2.base.atFlags & AC_ON) {
+            Collider_UpdateCylinder(&this->actor, &this->collider2);
+            this->collider2.dim.pos.y = this->actor.floorHeight;
+            CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->collider2.base);
+        }
+    }
+}
+
+s32 EnFamos_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                             Actor* thisx) {
+    EnFamos* this = THIS;
+
+    if (limbIndex == FAMOS_LIMB_BODY) {
+        Matrix_InsertTranslation(0.0f, 4000.0f, 0.0f, MTXMODE_APPLY);
+        Matrix_InsertZRotation_s(this->flipRot, MTXMODE_APPLY);
+        Matrix_InsertTranslation(0.0f, -4000.0f, 0.0f, MTXMODE_APPLY);
+
+    } else if (this->flippedTimer < 0 && // if set to -1, famos is dying
```

Comment:
```suggestion
    } else if ((this->flippedTimer < 0) && // if set to -1, famos is dying
```
This would probably also look better as 2 ifs.

Reply:
Why do you want to keep separating if conditions?

Reply:
It makes them easier to read than if they're one huge run-on line that gets split up. And it makes sense, if the things that are being checked are sufficiently different.